doc/pages/architecture/080-mqtt-v1-protocol.md

```diff
@@ -0,0 +1,278 @@
+#### Datastream Message (individual aggregation)
+
+Datastream messages for not object aggregated interfaces have a "v" key and an optional "t" key (which means timestamp) and they look like the followings:
+
+* `{"v": false}`
+
+* `{"v": 16.73}`
+
+* `{"v": 16.73, "t": "2018-09-20T13:17:02.890869Z"}`
+
+Timestamps are UTC timestamps (BSON 0x09 type), not ISO 8601 strings, when not provided reception timestamp is used.
+
+Previous payloads are in order BSON encoded as following hex dumps:
+
+```
+0000000 09 00 00 00 08 76 00 00 00
+```
+
+```
+0000000 10 00 00 00 01 76 00 7b 14 ae 47 e1 ba 30 40 00
+```
+
+```
+0000000 1b 00 00 00 09 74 00 2a 70 20 f7 65 01 00 00 01
+0000020 76 00 7b 14 ae 47 e1 ba 30 40 00
+```
+
+#### Datastream Message (object aggregation)
+
+Datastream messages (for interfaces with object aggregation) have a "v" key, that is a BSON subdocument, and an optional "t" key (which means timestamp) and they look like the following:
+
+* `{"v": {"temp": 25.3, "hum": 67.11}}`
+
+* `{"v": {"temp": 25.3, "hum": 67.11}, "t": "2018-09-20T14:08:34.811300Z"}`
+
+Timestamps are UTC timestamps (BSON 0x09 type), not ISO 8601 strings.
+
+Previous payloads are in order BSON encoded as following hex dumps:
+
+```
+0000000 28 00 00 00 03 76 00 20 00 00 00 01 68 75 6d 00
+0000020 d7 a3 70 3d 0a c7 50 40 01 74 65 6d 70 00 cd cc
+0000040 cc cc cc 4c 39 40 00 00
+```
+
+```
+0000000 33 00 00 00 09 74 00 fb 9d 4f f7 65 01 00 00 03
+0000020 76 00 20 00 00 00 01 68 75 6d 00 d7 a3 70 3d 0a
+0000040 c7 50 40 01 74 65 6d 70 00 cd cc cc cc cc 4c 39
+0000060 40 00 00
+```
+
+## Minimal Protocol
+
+A device might implement a subset of this protocol if needed. `/control/consumer/properties`, `/control/producer/properties` and `/emptyCache` might be ignored or not implemented if a device has no property interfaces.
+A further simplification might remove any requirement for any introspection message when previously provisioned, but this feature is not supported out of the box.
+
+## Error Handling
+
+A device might be forcefully disconnected due to any kind of error. Devices should wait a random ammount of time before trying to connect again to the broker.
+_session present_ might be also set to false to ensure a clean and consistent state (in that case messages such as _introspection_ and _empty cache_ should published as previously described).
+
+Malformed or unexpected messages are discarded and further actions might be taken.
+
+## Security
+
+Security and authentication features are not described here, they must be handled at a lower level. The MQTT broker must make sure that a certain device can publish data only for its realm/device id.
+This feature is accomplished with X509 client certificates and mutual ssl authentication, also the MQTT broker ensure that the Common Name matches the device topic (`<realm name>/<device id>`) when a new message is published.
+Certificate provisioning and renewal is handled out of band and it is not part of this protocol.
```

Comment:
This whole section must link to Pairing somehow, especially now that we fixed it.